src/app/FAKE/Program.fs

```diff
@@ -132,15 +132,16 @@ try
             match Boot.ParseCommandLine(cmdArgs) with
             | None ->
                 let buildScriptArg = if cmdArgs.Length > 1 && cmdArgs.[1].EndsWith ".fsx" then cmdArgs.[1] else Seq.head buildScripts
-                let fakeArgs = cmdArgs |> Array.filter (fun x -> x.StartsWith "-d:" = false)
-                let fsiArgs = cmdArgs |> Array.filter (fun x -> x.StartsWith "-d:") |> Array.toList
-                let args = CommandlineParams.parseArgs (fakeArgs |> Seq.filter ((<>) buildScriptArg) |> Seq.filter ((<>) "details"))
+                let fsiArgs, fakeArgs = cmdArgs |> Array.partition (fun x -> x.StartsWith "-d:")
```

Comment:
Now that '-pd' and '--print-details' are being expanded, we can't go check back in `cmdArgs` for 'details', we have to check the inflated list of arguments. That's why I removed the details filter here.  Then the lines below do the correct 'details' handling.